featurebyte/models/precomputed_lookup_feature_table.py

```diff
@@ -0,0 +1,291 @@
+    feature_store_model: FeatureStoreModel
+        Feature store
+    feature_table_id: PydanticObjectId
+        Id of the source feature table
+
+    Returns
+    -------
+    List[OfflineStoreFeatureTableModel]
+    """
+
+    feature_ids_set = set(feature_ids)
+    feature_lists = [
+        feature_list
+        for feature_list in feature_lists
+        if set(feature_list.feature_ids).intersection(feature_ids_set)
+    ]
+    feature_lists_relationships_info = _get_feature_lists_to_relationships_info(feature_lists)
+    primary_entity_ids = sorted(primary_entity_ids)
+
+    precomputed_lookup_feature_tables: Dict[
+        Tuple[EntityRelationshipInfo, ...], OfflineStoreFeatureTableModel
+    ] = {}
+    for feature_list in feature_lists:
+        for full_serving_entity_ids in feature_list.enabled_serving_entity_ids:
+            serving_entity_ids = EntityAncestorDescendantMapper.create(
+                feature_lists_relationships_info[feature_list.id],
+            ).keep_related_entity_ids(
+                entity_ids_to_filter=full_serving_entity_ids,
+                filter_by=primary_entity_ids,
+            )
+            lookup_steps = EntityLookupPlanner.generate_lookup_steps(
+                available_entity_ids=serving_entity_ids,
+                required_entity_ids=primary_entity_ids,
+                relationships_info=feature_lists_relationships_info[feature_list.id],
+            )
+            key = tuple(lookup_steps)
+            if not key:
+                continue
+            table = precomputed_lookup_feature_tables.get(key)
+            if table is None:
+                serving_names = [
+                    entity_id_to_serving_name[entity_id] for entity_id in serving_entity_ids  # type: ignore[index]
+                ]
+                table = OfflineStoreFeatureTableModel(
+                    name=get_precomputed_lookup_feature_table_name(
+                        feature_table_name, serving_names, lookup_steps
+                    ),
+                    feature_ids=[],
+                    primary_entity_ids=serving_entity_ids,
+                    serving_names=serving_names,
+                    entity_universe=EntityUniverseModel(
+                        query_template=SqlglotExpressionModel.create(
+                            get_child_entity_universe_template(
+                                lookup_steps=lookup_steps,
+                                entity_lookup_steps_mapping=entity_lookup_steps_mapping,
+                                feature_store=feature_store_model,
+                            )
+                        )
+                    ),
+                    precomputed_lookup_feature_table_info=PrecomputedLookupFeatureTableInfo(
+                        lookup_steps=lookup_steps,
+                        source_feature_table_id=feature_table_id,
+                    ),
+                    has_ttl=False,
```

Comment:
Is this always False?

Reply:
I think this should be set the same as the source feature table's ttl. Will fix.